File: JavaScript Front End/src/Messages/CommandMessages/GraphBuilding/AddSourceTaskCommandMsg.js
const AddSourceSinkTaskCommandMessage = require('./AddSourceSinkTaskCommandMsg.js').AddSourceSinkTaskCommandMessage;

class AddSourceTaskCommandMsg extends AddSourceSinkTaskCommandMessage
{
    constructor(taskName, taskTemporaryId, additionalInfo)
    {
        super(taskName, taskTemporaryId, additionalInfo);
        this.jsonMessage["@type"] = this.jsonMessage["@type"].concat("AddSourceTaskCommandMsg");
    }
}

module.exports = {AddSourceTaskCommandMsg: AddSourceTaskCommandMsg};

File: Java Back End/src/InfoSecCooker/Communication/Messages/CommandMessages/GraphBuilding/AddSourceTaskCommandMsg.java
package InfoSecCooker.Communication.Messages.CommandMessages.GraphBuilding;

public class AddSourceTaskCommandMsg extends AddSourceSinkTaskCommandMsg
{
    public AddSourceTaskCommandMsg(String taskName, long taskTemporaryId, String additionalInfo)
    {
        super(taskName, taskTemporaryId, additionalInfo);
    }
}
